### PR TITLE
 Refactor FUIAuth to split out email auth.

### PR DESCRIPTION
**Pull request by Yue-Wang-Google**

### CODE REVIEW

**Comment by Yue-Wang-Google** [comment]
@renkelvin

File: FirebaseEmailAuthUI/FUIPasswordVerificationViewController.m [hunk 1]
```diff
@@ -168,17 +170,19 @@ - (void)verifyPassword:(NSString *)password {
...
-        [self.authUI invokeResultCallbackWithAuthDataResult:authResult error:nil];
+        [self.authUI invokeResultCallbackWithAuthDataResult:authResult URL:nil error:nil];
       }];
     }];
   }];
 }
 
 - (IBAction)forgotPassword {
   UIViewController *viewController;
+  id<FUIAuthDelegate> delegate = self.authUI.delegate;
```
Comment (renkelvin) [inline FirebaseEmailAuthUI/FUIPasswordVerificationViewController.m]: Any specific reason for these changes?

File: FirebaseUI.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -7,14 +7,52 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
+		0A5A3B66FEDFCF1500BD7617 /* Pods_FirebaseFacebookAuthUI.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 7CE11F1C9A9C4891B13E660E /* Pods_FirebaseFacebookAuthUI.framework */; };
```
Comment (renkelvin) [inline FirebaseUI.xcodeproj/project.pbxproj]: Please use `pod deintegrate` to uncheck these cocoapods changes

File: samples/objc/FirebaseUI-demo-objc.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -7,6 +7,7 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
+		3547F58124B031899D39956A /* Pods_FirebaseUI_demo_objc.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 225BF144F30186355EC17E32 /* Pods_FirebaseUI_demo_objc.framework */; };
```
Comment (renkelvin) [inline samples/objc/FirebaseUI-demo-objc.xcodeproj/project.pbxproj]: `pod deintegrate`

File: samples/swift/FirebaseUI-demo-swift.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -35,6 +35,8 @@
 		C39BC0511DB812330060F6AF /* FUICustomPasswordVerificationViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = C39BC04F1DB812330060F6AF /* FUICustomPasswordVerificationViewController.swift */; };
 		C39BC0521DB812330060F6AF /* FUICustomPasswordVerificationViewController.xib in Resources */ = {isa = PBXBuildFile; fileRef = C39BC0501DB812330060F6AF /* FUICustomPasswordVerificationViewController.xib */; };
 		C3F23ECD1D80F3300020509F /* GoogleService-Info.plist in Resources */ = {isa = PBXBuildFile; fileRef = C3F23ECC1D80F3300020509F /* GoogleService-Info.plist */; };
+		E1ED177630EB565C69F6D623 /* Pods_FirebaseUI_demo_swiftTests.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 576EF7722D9EADB90A2BEED7 /* Pods_FirebaseUI_demo_swiftTests.framework */; };
```
Comment (renkelvin) [inline samples/swift/FirebaseUI-demo-swift.xcodeproj/project.pbxproj]: `pod deintegrate`

File: FirebaseEmailAuthUI/FUIEmailAuth.h [hunk 5]
```diff
@@ -0,0 +1,89 @@
+//
+//  Copyright (c) 2018 Email Inc.
```
Comment (morganchen12) [inline FirebaseEmailAuthUI/FUIEmailAuth.h]: @Yue-Wang-Google Please update the organization in your Xcode config

Comment (Yue-Wang-Google) [reply FirebaseEmailAuthUI/FUIEmailAuth.h]: I'm sorry. Pull request #544 is created to fix this.

Comment (morganchen12) [reply FirebaseEmailAuthUI/FUIEmailAuth.h]: Thanks, and no problem 😃

**Pull request merged**